packages/ui-contract-editor/src/lib/ContractEditor/index.js

```diff
@@ -72,24 +74,25 @@ const ContractEditor = (props) => {
     pasteToContract: props.pasteToContract
   };
 
-  const customElements = (attributes, children, element) => {
+  const customElements = (attributes, children, element, editor) => {
     const returnObject = {
       clause: () => (
-        <ClauseComponent
-          templateUri={element.data.src}
-          clauseId={element.data.clauseid}
-          clauseProps={props.clauseProps}
-          readOnly={props.readOnly}
-          {...attributes}
-        >
-            {children}
-        </ClauseComponent>
+          <ClauseComponent
+            templateUri={element.data.src}
+            clauseId={element.data.clauseid}
+            clauseProps={props.clauseProps}
+            readOnly={props.readOnly}
+            attributes={attributes}
+            editor={editor}
+          >
+              {children}
+          </ClauseComponent>
       ),
       variable: () => (
         <span id={element.data.id} {...attributes} className={VARIABLE}>{children}</span>
       ),
       conditional: () => (
-        <Conditional readOnly={props.readOnly} {...attributes}>{children}</Conditional>
+        <Conditional readOnly={props.readOnly} attributes={attributes}>{children}</Conditional>
```

Comment:
I'm curious why this needs to be this way, but it works in `variable` as `{...attributes}`

Reply:
variable is rendering an html element (span), while conditional is rendering a React component, so we should be passing attributes as a prop to the React component and then spreading attributes on the html element in that React component.

Reply:
Ah! This makes sense.